### DISCUSSION TITLE
Custom Link with Services

### DISCUSSION THREAD

**Discussion by brunodangelos** [netbox-community/netbox#8347] [Q&A] [answered]
Hello!
I'm a newby with Netbox and trying to create a custom link which uses a service ports assigned to a specific device. The content type i'm using is DCIM>device to show a link in this case. I am using this in the Link Text template field

{% if obj.service.name == 'ssh' %}  
ssh://{{obj.cf['user']}}@{{obj.primary_ip.address.ip}} -p {{obj.services.port}}
{% endif %}
but nothing shows up, what am i not doing right?

**Comment by mtinberg** [comment]
I didn't look deeply into this but note that in the first line you reference object.service.name​ and in the subsequent line object.services.name​ which are not the same.


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
From: brunodangelos ***@***.***>
Sent: Thursday, January 13, 2022 12:49 PM
To: netbox-community/netbox ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [netbox-community/netbox] Custom Link with Services (Discussion #8347)


Hello!
I'm a newby with Netbox and trying to create a custom link which uses a service ports assigned to a specific device. The content type i'm using is DCIM>device to show a link in this case. I am using this in the Link Text template field

{% if obj.service.name == 'ssh' %}
ssh://{{obj.cf['user']}}@{{obj.primary_ip.address.ip}} -p {{obj.services.name}}
{% endif %}
but nothing shows up, what am i not doing right?

—
Reply to this email directly, view it on GitHub<https://github.com/netbox-community/netbox/discussions/8347>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AAS7UMYJPO3PQHSWBI7CLB3UV4NDXANCNFSM5L4UBRWQ>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Comment by jeremystretch** [comment, marked as answer]
A device can have multiple services associated with it, so you presumably want to check for _any_ service named `ssh`. You could do something like this:

```
{% if object.services.filter(name='ssh').exists() %}
```

That will return True if there's a service named `ssh` associated with the device.

**Reply by brunodangelos** [reply]
It's worked to filter to ssh!
How i put port of filtered service object? Or I will need to create a new filter with filter service name, device and port? To make url add port of connection? Exemple: 

ssh://{{obj.cf['User']}}@{{obj.primary_ip.address.ip}} -p {{object of service port filtered}} <--

**Reply by jeremystretch** [reply]
Oh man, we're getting tricky now. :slightly_smiling_face: It should be something like this:

```
{% set service = obj.services.filter(name='ssh').first() %}
{% if service %}
ssh://{{ obj.cf.user }}@{{ obj.primary_ip.address.ip }}:{{ service.ports[0] }}
{% endif %}
```

**Reply by brunodangelos** [reply]
Worked, but this way he catch the first result of service, if have 2 services inserted he brings me the first if have name of ssh or not, i based of your help and created new link and worked.


{% if obj.services.filter(name='ssh').exists() and obj.services.filter(device=obj.id) %}
{% set service = obj.services.filter(name='ssh').first() %}
ssh://{{ obj.cf['Usuario'] }}@{{ obj.primary_ip.address.ip }}:{{ service.ports[0] }}
{% else %}
ssh://{{obj.cf['Usuario']}}@{{obj.primary_ip.address.ip}}:22
{% endif %}


Thank you for help!
